modules/shapefile/src/lib/parsers/parse-shapefile.js

```diff
@@ -12,7 +12,7 @@ export async function* parseShapefileInBatches(asyncIterator, options, context)
 
   // parse geometries
   const shapeIterator = await parseInBatches(asyncIterator, SHPLoader); // {shp: shx}
-  const shapeHeader = await shapeIterator.next();
+  const shapeHeader = (await shapeIterator.next()).value;
```

Comment:
Async generators return an object of `{value, done}` when called with `.next` directly

Reply:
yes but you have to check done before using value.

Reply:
True but this is the first batch of the `shapeIterator`, so it _should_ always be non-null as long as data exists. What's the better way to handle this? I just return `shapeHeader` along with each returned batch below